app/assets/javascripts/discourse/app/controllers/topic.js

```diff
@@ -931,8 +930,16 @@ export default class TopicController extends Controller.extend(
 
   @action
   jumpBottom() {
+    const highestPostNumber = this.model.highest_post_number;
+
+    if (document.getElementById(`post_${highestPostNumber}`)) {
+      // Do nothing when the last post is already rendered.
+      // This ensures the browser handles keyboard shortcuts like End.
+      return;
+    }
+
     // When a topic only has one lengthy post
-    const jumpEnd = this.model.highest_post_number === 1 ? true : false;
+    const jumpEnd = highestPostNumber === 1 ? true : false;
```

Comment:
if `highestPostNumber === 1`, won't it be in the DOM already? I think we don't need this check anymore because of the early return... wdyt?

Reply:
Yeah, that's a very good point. Removed that check. Thanks!